Upgrade CCM .NET Dependency Packages & Jenkins Plugins

ryanrichter94 opened the pull request:
## Description Of Changes
- Upgraded .NET CCM Dependency Packages (dotnet-aspnetcoremodule-v2, dotnet-8.0-runtime, and dotnet-8.0-aspnetruntime)
- Updates the Jenkins Plugins used by the Jenkins jobs

## Motivation and Context
Update components to deploy recent software for end users.

## Testing
1. Local Server 2025 VM

### Operating Systems Testing
- Windows Server 2019
- Windows Server 2022
- Windows Server 2025

## Change Types Made
* [ ] Bug fix (non-breaking change).
* [X] Feature / Enhancement (non-breaking change).
* [ ] Breaking change (fix or feature that could cause existing functionality to change).
* [ ] Documentation changes.
* [X] PowerShell code changes.

## Change Checklist
* [ ] Requires a change to the documentation.
* [ ] Documentation has been updated.
* [ ] Tests to cover my changes, have been added.
* [X] All new and existing tests passed?
* ~[ ] PowerShell code changes: PowerShell v3 compatibility checked?~ Not required in this repo.

## Related Issue

Fixes #306

ryanrichter94 committed:
Update Jenkins plugins to latest versions as of Nov. 14th 2025

ryanrichter94 committed:
Update .NET CCM Dependency packages to latest versions as of Nov. 14th 2025

ryanrichter94 commented:
Setting up a meeting with @JPRuskin on Monday to resolve the expired Azure test keys.

JPRuskin commented on files/jenkins.json:
```diff
@@ -1,33 +1,34 @@
-        { "name": "workflow-scm-step", "version": "437.v05a_f66b_e5ef8" },
-        { "name": "workflow-step-api", "version": "700.v6e45cb_a_5a_a_21" },
-        { "name": "workflow-support", "version": "968.v8f17397e87b_8" }
+        { "name": "workflow-api", "version": "1398.v67030756d3fb_" },
```
This version seems to fail to load on the latest LTS version of Jenkins, which causes issues.

```
2025-11-18 12:21:47.045+0000 [id=51]	SEVERE	jenkins.InitReactorRunner$1#onTaskFailed: Failed Loading plugin Pipeline: API v1398.v67030756d3fb_ (workflow-api)
java.io.IOException: Failed to load: Pipeline: API (workflow-api 1398.v67030756d3fb_)
[CRLF]>  - Jenkins (2.532) or higher required
	at hudson.PluginWrapper.resolvePluginDependencies(PluginWrapper.java:1020)
	at hudson.PluginManager$2$1$1.run(PluginManager.java:590)
	at org.jvnet.hudson.reactor.TaskGraphBuilder$TaskImpl.run(TaskGraphBuilder.java:175)
	at org.jvnet.hudson.reactor.Reactor.runTask(Reactor.java:304)
	at jenkins.model.Jenkins$5.runTask(Jenkins.java:1151)
	at org.jvnet.hudson.reactor.Reactor$2.run(Reactor.java:221)
	at org.jvnet.hudson.reactor.Reactor$Node.run(Reactor.java:120)
	at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:68)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
```

Version of Jenkins installed: `2.528.2`

JPRuskin replied:
```suggestion
        { "name": "workflow-api", "version": "1384.vdc05a_48f535f" },
```

Using this instead seems to work.

JPRuskin commented on files/jenkins.json:
```diff
@@ -1,33 +1,34 @@
+        { "name": "pipeline-groovy-lib", "version": "776.vfee5327b_b_a_5b_" },
+        { "name": "scm-api", "version": "712.v8846fdd68c88" },
+        { "name": "script-security", "version": "1385.v7d2d9ec4d909" },
+        { "name": "structs", "version": "353.v261ea_40a_80fb_" },
```
If we're having to make a change anyway, might as well catch the two updates that have been brought in since Friday:

```suggestion
        { "name": "structs", "version": "362.va_b_695ef4fdf9" },
```

JPRuskin commented on files/jenkins.json:
```diff
@@ -1,33 +1,34 @@
-        { "name": "script-security", "version": "1373.vb_b_4a_a_c26fa_00" },
-        { "name": "structs", "version": "343.vdcf37b_a_c81d5" },
+        { "name": "mailer", "version": "522.va_995fa_cfb_8b_d" },
+        { "name": "pipeline-groovy-lib", "version": "776.vfee5327b_b_a_5b_" },
```
```suggestion
        { "name": "pipeline-groovy-lib", "version": "787.ve2fef0efdca_6" },
```

JPRuskin committed:
(#306) Adds Jenkins tests to ensure Jobs and Plugins are present

In some circumstances, the tests we had would not accurately show that Jenkins had successfully loaded jobs and plugins.

These tests aim to validate that:
- The Jenkins plugins have all been installed successfully, and are loaded correctly.
- The Jenkins jobs have all been loaded correctly, and are available for use.

This should allow our automated testing to better reflect the state of the environment.

JPRuskin committed:
(#306) Update Jenkins plugins to latest versions as of Nov. 18th 2025

A few plugins were slightly outdated. This brings them up to the latest version supported by Jenkins LTS.

JPRuskin committed:
(maint) Fix Jenkins Plugin Download URL

Some users were seeing issues downloading the from the existing URL. This should work better.

JPRuskin reviewed (approved):
Ryan has reviewed the changes I made, I agree with the changes he made - LGTM.
